File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/ui/demo/splash/presenter/SplashPresenter.java
package com.github.alexeygorovoy.moxytemplate.ui.demo.splash.presenter;

import com.arellomobile.mvp.InjectViewState;
import com.github.alexeygorovoy.moxytemplate.ui.common.moxy.BaseMvpPresenter;
import com.github.alexeygorovoy.moxytemplate.ui.demo.splash.view.SplashView;
import com.github.alexeygorovoy.moxytemplate.rx.RxSchedulers;

import java.util.concurrent.TimeUnit;

import rx.Single;
import rx.Subscription;
import timber.log.Timber;


@InjectViewState
public class SplashPresenter extends BaseMvpPresenter<SplashView> {

    private final RxSchedulers rxSchedulers;

    public SplashPresenter(RxSchedulers schedulers) {
        this.rxSchedulers = schedulers;
    }

    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
        // real loading will happen here if needed
        Subscription sub = Single.just("")
                .delay(500, TimeUnit.MILLISECONDS)
                .compose(rxSchedulers.computationToMainSingle())
                .subscribe(
                        it -> {
                            getViewState().showSuccessToast("App started!");
                            getViewState().openHeroesScreen();
                        },
                        throwable -> Timber.e(throwable, "error on splash!")
                );
        unsubscribeOnDestroy(sub);
    }

}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/ui/demo/splash/presenter/SplashPresenter.kt
package com.github.alexeygorovoy.moxytemplate.ui.demo.splash.presenter

import com.arellomobile.mvp.InjectViewState
import com.github.alexeygorovoy.moxytemplate.ui.common.moxy.BaseMvpPresenter
import com.github.alexeygorovoy.moxytemplate.ui.demo.splash.view.SplashView
import com.github.alexeygorovoy.moxytemplate.utils.rx.RxSchedulers
import rx.Single
import timber.log.Timber
import java.util.concurrent.TimeUnit

@InjectViewState
class SplashPresenter(private val rxSchedulers: RxSchedulers) : BaseMvpPresenter<SplashView>() {

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()
        // real loading will happen here if needed
        Single.just("")
            .delay(500, TimeUnit.MILLISECONDS)
            .compose(rxSchedulers.computationToMainSingle())
            .subscribe(
                {
                    viewState.showSuccessToast("App started!")
                    viewState.openHeroesScreen()
                },
                { throwable -> Timber.e(throwable, "error on splash!") }
            )
            .unsubscribeOnDestroy()
    }
}
